### PR TITLE
consolidating some of the error checks

### PR DESCRIPTION
**Pull request by JackStat**
Picking up some low hanging fruit here to get myself familiar with the package.  I added tests for my modifications as well

### CODE REVIEW

**Comment by Neil-Schneider** [comment]
There are a lot of extra/blank lines added in these commits. With proper indentation, which you have, it should not be necessary to add these blank lines to further offset code block. They don't substancially increase readability and increase the length of code base. (If this was consistantly done throughout the entire code base, the number of code lines would be 2-3 times longer than necessary.)

File: R/checks.R [hunk 1]
```diff
@@ -49,4 +59,46 @@ getVarNames <- function(x){
...
+  }
+  
+}
+
+checkVarType <- function(x, y){
+  
+  nms <- getVarNames(x)
+  
+  # Excessive Factors
+  Factors <- sapply(x, function(X){is.factor(X)})
```
Comment (pdmetcalfe) [inline R/checks.R]:
``` r
Factors <- vapply(x, is.factor, TRUE)
```

File: R/checks.R [hunk 2]
```diff
@@ -49,4 +59,46 @@ getVarNames <- function(x){
...
+  nLevels <- sapply(x, function(X){length(levels(X))})
```
Comment (pdmetcalfe) [inline R/checks.R]:
``` r
nLevels <- vapply(x, nlevels, 0L)
```

**Comment by pdmetcalfe** [comment]
Thanks for this - I'm _really_ happy to get pull requests, with associated unit tests, that tidy gbm up! [dear rest of world, please note!]

I don't think we have a "whitespace" policy, and I'm not terribly keen to develop one, either. :smile:

Could you possibly have a look at the comments I put in your code - see if they make sense?

**Comment by JackStat** [comment]
@pdmetcalfe Thanks for recommending vapply. I incorporated your changes. That for loop that sets up the var.type and var.level objects is next on my list.  @Neil-Schneider My apologies for the extra whitespace but it does improve the readability for me and I was trying to make the formatting consistent inside of that for loop.

**Pull request merged**